CMSgov/cms-mdct-qmr
# Add .githooks and pre-commit hook

## Purpose

Create a pre-commit hook that runs our Prettier linter before files are commited.

## Approach

Standard Git hooks are stored in `.git/hooks`, a directory not managed by source control. This PR creates a directory `.githooks` that will be included in source control, adds a command to the `dev` script to move the contents to `.git/hooks`, and adds a simple hook to run the Prettier linter.

Any future hooks or changes to the `.githooks` directory will be reflected in `.git/hooks` whenever the user runs the `./dev local` command.

## Testing

- Briefly run `./dev local` to make sure hooks are copied to `.git/hooks`
  - You can cancel the command as soon as it starts installing packages
- Make an egregious change to a file that should be watched by Prettier. (e.g. add a bunch of whitespace to App.tsx)
- Stage the change
- Manually run the pre-commit hook `bash .git/hooks/pre-commit`
- Confirm the egregious change is less egregious

#### Pull Request Creator Checklist


#### Pull Request Reviewer/Assignee Checklist
